ksqldb-common/src/main/java/io/confluent/ksql/reactive/BasePublisher.java

```diff
@@ -65,8 +66,9 @@ public void subscribe(final Subscriber<? super T> subscriber) {
     }
   }
 
-  public void close() {
+  public Future<Void> close() {
     ctx.runOnContext(v -> doClose());
+    return Future.succeededFuture();
```

Comment:
Did the signature for `Publisher` change?  

Can line 70 fail in any interesting way?

Reply:
`Publisher` is super simple:
```
public interface Publisher<T> {
  void subscribe(Subscriber<? super T> var1);
}
```

The reason I made this change is because `HttpConnection#close` became asynchronous. 

That caused some of our tests to fail (specifically the `scalablePushBandwidthThrottleTestHTTP2`) because it assumed that the connection was closed before doing some other stuff. This allowed me to attach a `CountdownLatch` to the close callback.

Reply:
> Can line 70 fail in any interesting way?

Good question. It can but we ignore it - so it's trivially "successful". This is all it does:
```
  protected void sendComplete() {
    try {
      sentComplete = true;
      // If a publisher immediately completes, we allow it not to yet have a subscriber, which will
      // get a complete message upon subscription.
      if (subscriber != null) {
        subscriber.onComplete();
      }
    } catch (Exception ex) {
      logError("Exception encountered in onComplete", ex);
    }
  }
```